python/tests/operations/join.py

```diff
@@ -275,6 +275,55 @@ def test_different_names(self):
 
         self.assertTrue(np.equal(joined_labels.tolist(), ref).all())
 
+    def test_different_names_different_length(self):
+        """Test Label joining using labels with different names and different length."""
+        property_labels_1 = Labels(
+            ("structure", "prop_1"), np.vstack(2 * [np.arange(5)]).T
+        )
+        property_labels_2 = Labels(
+            ("structure", "prop_2", "prop_3"), np.vstack(3 * [np.arange(5)]).T
+        )
+
+        block_1 = TensorBlock(
+            values=np.zeros([2, 5]),
+            samples=self.sample_labels,
+            components=[],
+            properties=property_labels_1,
+        )
+
+        block_2 = TensorBlock(
+            values=np.zeros([2, 5]),
+            samples=self.sample_labels,
+            components=[],
+            properties=property_labels_2,
+        )
+
+        joined_ts = eop.join(
+            [TensorMap(self.keys, [block_1]), TensorMap(self.keys, [block_2])],
+            axis="properties",
+        )
+
+        joined_labels = joined_ts.block(0).properties
+
+        self.assertEqual(joined_labels.names, ("tensor", "property"))
+
+        ref = np.array(
```

Comment:
why do you get 2 properties when you merge 3 properties with 2 properties where only 1 property has the same name?

Reply:
There is a way to improve sure. We could try to only replace those properties that are not the same, but I resolving name clashes is already a bit messy. If you have ideas I am open for suggestions.

Reply:
okay seems like a discussion for another PR